packages/ember-cli-eyeglass/src/index.ts

```diff
@@ -70,26 +70,27 @@ function isLazyEngine(addon: any): boolean {
 }
 
 //eslint-disable-next-line @typescript-eslint/no-explicit-any
-function isUsingEmbroider(app: any): boolean {
+function hasEmbroiderDependency(app: any): boolean {
```

Comment:
Sounds good. Think it's worth keeping the dependency checks and adding a warning if the addon detects a de-sync? (e.g.: The app has a dep on `embroider/core` but hasn't set the flag)

I'm considering this because the downstream build errors that you could get from this being misconfigured can be cryptic and difficult to trace back to `ember-cli-eyeglass`.

Reply:
nvm, I realized that this check wouldn't actually catch the situation I'm solving with this PR.

Alright, updated diff to remove the auto-detect code.